### PR TITLE
Update pre-release CI workflow

### PR DESCRIPTION
**Pull request by tylerjw**
### Description

This uses this github action https://github.com/marketplace/actions/publish-unit-test-results to publish nice looking test results.  This also cleans up the readme badges for CI to link to the correct workflows.

### CODE REVIEW

**Comment by tylerjw** [comment]
After more careful reading I discovered this only works if it is run on the repo and not on a pull request from a fork.  I'm trying adding it to the pre-release test and I fixed the fork of industrial_ci we use to be back on master.

**Comment by tylerjw** [comment]
``` 2021-04-30 17:16:52 +0000 - github_action - WARNING - This action is running on a pull_request event for a fork repository. It cannot do anything useful like creating check runs or pull request comments.
Warning: This action is running on a pull_request event for a fork repository. It cannot do anything useful like creating check runs or pull request comments.```

**Comment by tylerjw** [comment]
Testing here: https://github.com/tylerjw/moveit/runs/2477750664?check_suite_focus=true

**Comment by tylerjw** [comment]
It seems that pre-release industrial_ci jobs are succeeding when they should be failing: https://github.com/ros-planning/moveit/runs/2476916607

If you look at the raw logs you'll see that it fails but industrial_ci fails to detect that becasue colcon isn't installed.  I created this issue for this: https://github.com/ros-industrial/industrial_ci/issues/666

**Comment by tylerjw** [comment]
Testing here: https://github.com/ros-planning/moveit/pull/2638

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -81,11 +81,11 @@ jobs:
...
             ccache-${{ env.CACHE_PREFIX }}
       - name: industrial_ci
-        uses: tylerjw/industrial_ci@clang-tidy-modified-filter
+        uses: ros-industrial/industrial_ci@master
         env: ${{ matrix.env }}
-      - name: upload test artifacts (on failure)
+      - name: upload test artifacts
         uses: actions/upload-artifact@v2
-        if: failure()
+        if: always()
```
Comment (rhaschke) [inline .github/workflows/ci.yaml:88]: Why do you want to upload test results always now?

Comment (tylerjw) [reply .github/workflows/ci.yaml:88]: That's left over from me trying to use the that report action.  I'll revert that.

Comment (tylerjw) [reply .github/workflows/ci.yaml]: fixed

**Comment by tylerjw** [comment]
@rhaschke the reason I didn't bloom moveit for noetic here or either of the versions of moveit_resources was that I have had trouble getting the pre-release tests to actually pass (however they look like they pass if you allow them to use colcon instead of some other builder because of the issue with colcon and the buildfarm).

**Comment by tylerjw** [comment]
Here is a link to me testing pre-release of moveit on melodic: https://github.com/tylerjw/moveit/runs/2493888794

**Comment by tylerjw** [comment]
I think one of the pilz tests may have gone flaky.  :cry: 
```
&amp;#x001B;[31m[integrationtest_sequence_action_with_gripper-4] process has died [pid 72677, exit code -6, cmd /home/runner/work/moveit/moveit/.work/target_ws/devel/.private/pilz_industrial_motion_planner/lib/pilz_industrial_motion_planner/integrationtest_sequence_action --gtest_output=xml:/home/runner/work/moveit/moveit/.work/target_ws/build/pilz_industrial_motion_planner/test_results/pilz_industrial_motion_planner/rosunit-integrationtest_sequence_action_with_gripper.xml __name:=integrationtest_sequence_action_with_gripper __log:=/root/.ros/log/9fcb4614-ac3c-11eb-b1e6-0242ac110002/integrationtest_sequence_action_with_gripper-4.log].```

**Comment by rhaschke** [comment]
> I think one of the pilz tests may have gone flaky.

@jschleicher, could you have a look please?

**Comment by tylerjw** [comment]
Noetic just passed pre-release with this.

> Summary: 1578 tests, 0 errors, 0 failures, 2 skipped

I will go ahead and bloom the noetic release tomorrow morning if no one objects.

**Comment by tylerjw** [comment]
Re-running CI on this.

**Comment by rhaschke** [comment]
@tylerjw, do you have an idea why the `prerelease` test passes here, but not if `moveit` is the downstream for `moveit_resources`?

**Comment by jschleicher** [comment]
> > I think one of the pilz tests may have gone flaky.
> 
> @jschleicher, could you have a look please?

@rhaschke Could you provide more details, please? @tylerjw What did you change to make it pass now?

**Comment by tylerjw** [comment]
> @tylerjw What did you change to make it pass now?

I changed nothing, I just re-ran the test and it passed.  That is why I believe it has gone flaky in some way.

**Comment by jschleicher** [comment]
Do you have more logging output than in https://github.com/ros-planning/moveit/pull/2638#issuecomment-831452577 ?
```
&amp;#x001B;[31m[integrationtest_sequence_action_with_gripper-4] process has died [pid 72677, exit code -6, cmd /home/runner/work/moveit/moveit/.work/target_ws/devel/.private/pilz_industrial_motion_planner/lib/pilz_industrial_motion_planner/integrationtest_sequence_action --gtest_output=xml:/home/runner/work/moveit/moveit/.work/target_ws/build/pilz_industrial_motion_planner/test_results/pilz_industrial_motion_planner/rosunit-integrationtest_sequence_action_with_gripper.xml __name:=integrationtest_sequence_action_with_gripper __log:=/root/.ros/log/9fcb4614-ac3c-11eb-b1e6-0242ac110002/integrationtest_sequence_action_with_gripper-4.log].
```
> I changed nothing, I just re-ran the test and it passed. That is why I believe it has gone flaky in some way.

**Review by rhaschke — APPROVED** [review]

**Pull request merged**